libminifi/test/keyvalue-tests/PersistableKeyValueStoreServiceTest.cpp

```diff
@@ -0,0 +1,257 @@
+    persistable_key_value_store_service_node = process_group->findControllerService("testcontroller");
+    REQUIRE(persistable_key_value_store_service_node != nullptr);
+    persistable_key_value_store_service_node->enable();
+
+    controller = std::dynamic_pointer_cast<minifi::controllers::PersistableKeyValueStoreService>(
+        persistable_key_value_store_service_node->getControllerServiceImplementation());
+    REQUIRE(controller != nullptr);
+  }
+
+ protected:
+  char *state_dir;
+  std::shared_ptr<minifi::Configure> configuration;
+  std::shared_ptr<core::Repository> test_repo;
+  std::shared_ptr<core::Repository> test_flow_repo;
+  std::shared_ptr<core::ContentRepository> content_repo;
+  std::shared_ptr<minifi::io::StreamFactory> stream_factory;
+
+  std::unique_ptr<core::YamlConfiguration> yaml_config;
+  std::unique_ptr<core::ProcessGroup> process_group;
+
+  std::shared_ptr<core::controller::ControllerServiceNode> persistable_key_value_store_service_node;
+  std::shared_ptr<minifi::controllers::PersistableKeyValueStoreService> controller;
+
+  TestController testController;
+};
+
+
+TEST_CASE_METHOD(PersistableKeyValueStoreServiceTestsFixture, "PersistableKeyValueStoreServiceTestsFixture set and get", "[basic]") {
+  const char* key = "foobar";
+  const char* value = "234";
+  REQUIRE(true == controller->set(key, value));
+
+  SECTION("without persistence") {
+  }
+  SECTION("with persistence") {
+    controller->persist();
+    loadYaml();
+  }
+
+  std::string res;
+  REQUIRE(true == controller->get(key, res));
+  REQUIRE(value == res);
+}
+
+TEST_CASE_METHOD(PersistableKeyValueStoreServiceTestsFixture, "PersistableKeyValueStoreServiceTestsFixture special characters", "[basic]") {
+  const char* key = "[]{}()==\\=\n\n";
+  const char* value = ":./'\\=!\n=[]{}()";
+  REQUIRE(true == controller->set(key, value));
+
+  SECTION("without persistence") {
+  }
+  SECTION("with persistence") {
+    controller->persist();
+    loadYaml();
+  }
+
+  std::string res;
+  REQUIRE(true == controller->get(key, res));
+  REQUIRE(value == res);
+}
+
+TEST_CASE_METHOD(PersistableKeyValueStoreServiceTestsFixture, "PersistableKeyValueStoreServiceTestsFixture set and get all", "[basic]") {
+  const std::unordered_map<std::string, std::string> kvs = {
+      {"foobar", "234"},
+      {"buzz", "value"},
+  };
+  for (const auto& kv : kvs) {
+    REQUIRE(true == controller->set(kv.first, kv.second));
+  }
+
+  SECTION("without persistence") {
+  }
+  SECTION("with persistence") {
+    controller->persist();
+    loadYaml();
+  }
+
+  std::unordered_map<std::string, std::string> kvs_res;
+  REQUIRE(true == controller->get(kvs_res));
+  REQUIRE(kvs == kvs_res);
+}
+
+TEST_CASE_METHOD(PersistableKeyValueStoreServiceTestsFixture, "PersistableKeyValueStoreServiceTestsFixture set and overwrite", "[basic]") {
+  const char* key = "foobar";
+  const char* value = "234";
+  const char* new_value = "baz";
+  REQUIRE(true == controller->set(key, value));
```

Comment:
Comparing a boolean expression to `true` is redundant.

Reply:
Yep, but in tests I think it is better readable.